tools/plugin-base/src/main/java/io/spine/tools/gradle/ConfigurationName.java

```diff
@@ -41,84 +31,125 @@
+     *
+     * <p>Users cannot add dependencies directly to this configuration. However, this configuration
+     * may be resolved.
      */
-    TEST_COMPILE_CLASSPATH(TEST_COMPILE_CLASSPATH_CONFIGURATION_NAME),
+    compileClasspath,
 
     /**
-     * The {@code testRuntimeOnly} configuration.
+     * Dependencies available at runtime but not at compile-time.
+     *
+     * <p>Suitable for SPI implementations loaded via {@link java.util.ServiceLoader} or other
+     * classpath scanning utilities.
      */
-    TEST_RUNTIME_ONLY(TEST_RUNTIME_ONLY_CONFIGURATION_NAME),
+    runtimeOnly,
 
     /**
-     * The {@code implementation} configuration.
+     * All the dependencies included for the Java module runtime.
+     *
+     * <p>Users cannot add dependencies directly to this configuration. However, this configuration
+     * may be resolved.
      */
-    IMPLEMENTATION(IMPLEMENTATION_CONFIGURATION_NAME),
+    runtimeClasspath,
 
     /**
-     * The {@code runtimeClasspath} configuration.
+     * The annotation processors used during the compilation of the tests of this module.
+     *
+     * <p>These dependencies are not accessible to the user at compile-time or at runtime directly.
      */
-    RUNTIME_CLASSPATH(RUNTIME_CLASSPATH_CONFIGURATION_NAME),
+    testAnnotationProcessor,
 
     /**
-     * The {@code testRuntimeClasspath} configuration.
+     * Dependencies on which the Java module tests rely for implementation.
+     *
+     * <p>The dependencies are available at compile-time of the test code and at the test runtime.
      */
-    TEST_RUNTIME_CLASSPATH(TEST_RUNTIME_CLASSPATH_CONFIGURATION_NAME),
+    testImplementation,
 
     /**
-     * A custom configuration for downloading artifacts from repositories.
+     * All the dependencies included for the Java module tests compilation.
+     *
+     * <p>Users cannot add dependencies directly to this configuration. However, this configuration
+     * may be resolved.
      */
-    FETCH("fetch"),
+    testCompileClasspath,
 
     /**
-     * The {@code compile} configuration.
+     * Dependencies available at test runtime but not at compile-time.
+     *
+     * <p>For example, JUnit runners may be depended on with this configuration.
      */
-    @SuppressWarnings("deprecation")
-    // Required in order to add Protobuf dependencies.
-    // See issue https://github.com/google/protobuf-gradle-plugin/issues/242.
-    COMPILE(COMPILE_CONFIGURATION_NAME);
+    testRuntimeOnly,
 
-    private final String value;
+    /**
+     * All the dependencies included for the Java module test runtime.
+     *
+     * <p>Users cannot add dependencies directly to this configuration. However, this configuration
+     * may be resolved.
+     */
+    testRuntimeClasspath,
 
-    ConfigurationName(String value) {
-        this.value = value;
-    }
+    /**
+     * {@code protobuf} extension block exposed by the Gradle Protobuf plugin.
```

Comment:
That's not the case. Those two elements of the build model have the same name, though are independent. 

What this doc should say is that this is a configuration, which allows compiling `.proto` files from the dependencies. Also, please mention that the users should use `compile` when it is only required to add some Protobuf definition into the Protobuf compilation "classpath".